File: yacs-admin/app/models/section.rb
class Section < ActiveResource::Base
  
  self.site="https://127.0.0.1/api/v5/"
  belongs_to :course
  attr_accessor :name
  attr_accessor :crn
  attr_accessor :seats
  attr_accessor :seats_taken
  attr_accessor :instructors
  attr_accessor :num_periods
  attr_accessor :course_id

end

File: yacs-admin-angular/src/app/section/section.ts
/* Section class
 * Based off of: https://github.com/YACS-RCOS/yacs-web/blob/master/src/app/course-list/section/section.ts
 */

import { Period } from './period';

export class Section {
  id: number;
  course_id: number;
  name: string;
  crn: number;
  instructors: string[];
  seats: number;
  seats_taken: number;
  conflicts: number[];
  //periods: Period[];
  num_periods: number;

  constructor(id, course_id, name, crn, instructors, seats, seats_taken, conflicts, periods,
     num_periods, course_name, course_number, dep_code) {
    this.id = id;
    this.course_id = course_id;
    this.name = name;
    this.instructors = instructors;
    this.seats = seats;
    this.seats_taken = seats_taken;
    this.conflicts = conflicts;
    //this.periods = periods;
    this.num_periods = num_periods;
  }
}
